### PR TITLE
Make Room Avaiable APIs And Fixed all comments

### PR DESCRIPTION
**Pull request by harmeshverma01**

### CODE REVIEW

File: room/views.py [hunk 1]
```diff
@@ -12,10 +12,10 @@ class RoomView(APIView):
     permission_classes = [admin_required]
     
     def get(self, request, id=None):
-        user = Room.objects.all()
-        price = request.GET.get('price', None)
-        if price is not None:
-            user = user.filter(price=price)
+        user = Room.objects.filter(price__range = ['1000', '5000'] )
```
Comment (TACGithub) [inline room/views.py]: why are these hard-coded?

File: room/views.py [hunk 2]
```diff
@@ -25,10 +25,7 @@ class RoombookedView(APIView):
     permission_classes = [admin_required]
     
     def get(self, request, id=None):
-        user = Room.objects.all()
-        date = request.GET.get('date', None)
-        if date is not None:
-            user = user.filter(date=date)
+        user = Room.objects.filter(date__range=['2011-01-01', '2011-01-31'])
```
Comment (TACGithub) [inline room/views.py]: Hardcoded here also, and why the variable name is user if we are getting rooms in the queryset.

File: room/views.py [hunk 3]
```diff
@@ -63,6 +60,20 @@ def delete(self, request, id=None):
        return Response(({"message": "Room is deleted"}),status=status.HTTP_204_NO_CONTENT)                
 
 
+
+class RoomAvaiableView(APIView):
+    serializer_class = RoomSerializer
+    
+    def get(self, request, id=None):
+        user = Room.objects.all()
```
Comment (TACGithub) [inline room/views.py]: variable name

File: room/views.py [hunk 4]
```diff
@@ -63,6 +60,20 @@ def delete(self, request, id=None):
...
+        date = request.GET.get('date', None)
```
Comment (TACGithub) [inline room/views.py]: No date range for filter.

File: room/models.py [hunk 5]
```diff
@@ -10,8 +11,8 @@ class Room(models.Model):
     image = models.ImageField()
     floor = models.IntegerField()
     user = models.ForeignKey(to=User, on_delete=models.CASCADE, related_name='User', null=True)
-    bad_required = models.IntegerField(null=True)
-    date = models.DateTimeField(null=True)
+    bad_count = models.IntegerField(null=True)
+    date = models.DateTimeField(auto_now=False, null=True)
```
Comment (TACGithub) [inline room/models.py]: what does this date referring to?

File: room/models.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+from enum import auto
```
Comment (TACGithub) [inline room/models.py]: remove this import

File: room/views.py (hunk 1 above)
Comment (harmeshverma01) [reply room/views.py]: fixed

File: room/views.py (hunk 2 above)
Comment (harmeshverma01) [reply room/views.py]: fixed

File: room/views.py (hunk 3 above)
Comment (harmeshverma01) [reply room/views.py]: fixed

File: room/views.py (hunk 4 above)
Comment (harmeshverma01) [reply room/views.py]: fixed

File: room/models.py (hunk 5 above)
Comment (harmeshverma01) [reply room/models.py]: fixed

File: room/models.py (hunk 6 above)
Comment (harmeshverma01) [reply room/models.py]: fixed

**Pull request merged**